### PR TITLE
[FIX] website_blog: Change opacity for cover images

### PR DESCRIPTION
**Pull request by pedrobaeza**
Switch opacity value for posts, as now in v10 the meaning of that
property is for an upper solid frame, not for the image itself:

* opacity=1.0 in v9 means solid image
* opacity=1.0 in v10 means solid color frame.

So what we do is to get the inverse percentage of the opacity: abs(1 -opacity).

@Tecnativa

### CODE REVIEW

**Comment by OSevangelist** [comment]
I haven't tested it with an actual migration but that change seems reasonable (and after all relatively small) to me. So +1 from me for merging the PR

**Review by cubells — APPROVED** [review]
:+1: 

LGTM

File: addons/website_blog/migrations/10.0.1.0/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+from odoo.tools import safe_eval
+
+
+def convert_blog_post_cover(env):
+    """Switch opacity value for posts, as now in v10 the meaning of that
+    property is for an upper solid frame, not for the image itself.
+    """
+    post_obj = env['blog.post'].with_context(active_test=False)
+    for post in post_obj.search([('cover_properties', '!=', False)]):
+        props = safe_eval(post.cover_properties)
```
Comment (hbrunn) [inline addons/website_blog/migrations/10.0.1.0/post-migration.py:15]: upstream uses JSON here, so I think you should do the same: https://github.com/OCA/OCB/blob/10.0/addons/website_blog/controllers/main.py#L166

File: addons/website_blog/migrations/10.0.1.0/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        if 'opacity' in props:
+            props['opacity'] = str(abs(1 - float(props['opacity'])))
+        post.cover_properties = str(props).replace("'", '"')
```
Comment (hbrunn) [inline addons/website_blog/migrations/10.0.1.0/post-migration.py:18]: and then here dumps

**Comment by pedrobaeza** [comment]
OK, you're right. I'll do the PR changing it.

**Pull request merged**